lib/kino/proxy.ex

```diff
@@ -0,0 +1,81 @@
+defmodule Kino.Proxy do
+  @moduledoc """
+  Functionality for handling proxy requests forwarded from Livebook.
+
+  Livebook proxies requests at the following paths:
+
+    * `/sessions/:id/proxy/*path` - a notebook session
+
+    * `/apps/:slug/:session_id/proxy/*path` - a specific app session
+
+    * `/apps/:slug/proxy/*path` - generic app path, only supported for
+      single-session apps
```

Comment:
```suggestion
    * `/apps/:slug/proxy/*path` - generic app path, only supported for
      single-session apps. If the application has automatic shutdows
      enabled and it is not currently available, it will be automatically started
```